docs/developing.md

```diff
@@ -1,58 +1,54 @@
 # Development guide
 
-This guide explains how to make changes to the `frontend-shared` package and use them with your local copy of the [Hypothesis client](https://github.com/hypothesis/client).
+## Development workflow
 
-## Client setup
-
-Follow these steps to set up `frontend-shared` in the client repository.
-
-#### 1. Install yalc
-
-Install the `yalc` tool globally if you have not already done so.
+Run a local webserver that will rebuild on changes to JS or SASS source.
 
 ```shell
-yarn global add yalc
+$ make dev
 ```
 
-_Note_: [yalc](https://www.npmjs.com/package/yalc) is the tool that we use for locally making changes to a shared package and testing the changes in another project.
+You can view the pattern library at http://localhost:4001/ui-playground
 
-#### 2. Build & publish
+## Testing locally with other projects
 
-Build the package and then publish to your local yalc repository.
+This guide explains how to test other applications (e.g. [`client`](https://github.com/hypothesis/client), `lms`) with
+a local build of this project's package.
 
-```shell
-# from the frontend-shared dir
-$ yarn build
-$ yalc publish
-```
+### Configure
 
-#### 3. Link package
+1. Install [`yalc`](https://www.npmjs.com/package/yalc)
+2. Build and publish locally to `yalc`
 
-In the `client` repository, link to the package published with yalc.
+   ```shell
+   $ cd frontend-shared
+   $ yarn build
+   $ yalc publish
+   ```
 
-```shell
-# from the client dir
-$ yalc link @hypothesis/frontend-shared
-```
+3. Link to this local package from the target application
 
-This will install the built local version of your `frontend-shared` repository in the `client`
-without a using symlink.
+   ```shell
+   $ cd <application project directory>
+   $ yalc link @hypothesis/frontend-shared
+   ```
 
-_Note_: Currently, there is no support for automatically rebuilding so any changes made to the `frontend-shared` files will need to be rebuilt and re-pushed as follows:
+### Develop
+
+To push changes from `frontend-shared` to `yalc` installations:
 
 ```shell
-# from the frontend-shared dir
+$ cd frontend-shared
 $ yarn build
-$ yalc push
+$ yarn push
 ```
 
-_Note_: `yalc push` will automatically "push" the updated files to all copies of `frontend-shared` that have been linked via `yalc link @hypothesis/frontend-shared`.
-
-### Removing the link
+### Clean up
 
-If you wish to revert back to the current installed version of `frontend-shared`, simply run:
+Remove `yalc` linking from an application and restore the project-defined
+published package version:
 
 ```shell
-# from the client dir
-$ yarn --check-files
+$ cd <application project directory>
+$ yalc remove @hypothesis/frontend-shared
```

Comment:
This is the most salient change. I'm not convinced `yarn --check-files` does what this intends; I am more certain that `yalc remove` _does_.

Reply:
It looks to me like `yalc remove` and `yarn --check-files` do different things. `yalc remove` removes the copy of the package in the project's `.yalc` dir, but does't appear to update the contents of `node_modules/`, which is what matters most. `yarn --check-files` ensures that the contents of `node_modules/` matches `yarn.lock` (ie. reset all files to published packages from npm) but doesn't touch the `.yalc` directory.

In other words, it doesn't look like `yalc remove` "restores the project-defined package version".

```sh
# Initial state - `@hypothesis/frontend-shared` installed from npm
$ ls -lh node_modules/@hypothesis/ | grep frontend-shared
drwxr-xr-x  8 robert  staff   256B 26 Jul 08:52 frontend-shared

# After running `yarn link`, `node_modules/hypothesis-shared` becomes a symlink
$ yalc link @hypothesis/frontend-shared
Package @hypothesis/frontend-shared@3.6.0+c970582d linked ==> /Users/robert/hypothesis/client/node_modules/@hypothesis/frontend-shared
$ ls -lh node_modules/@hypothesis/ | grep frontend-shared
lrwxr-xr-x  1 robert  staff    65B 26 Jul 08:52 frontend-shared -> /Users/robert/hypothesis/client/.yalc/@hypothesis/frontend-shared
$ ls -lh .yalc/@hypothesis/
total 0
drwxr-xr-x  9 robert  staff   288B 26 Jul 08:52 frontend-shared

# Running `yalc remove` removes the copy from `.yalic/`, but doesn't update the symlink in `node_modules/`
$ yalc remove @hypothesis/frontend-shared
Removing installation of @hypothesis/frontend-shared in /Users/robert/hypothesis/client
$ ls -lh node_modules/@hypothesis/ | grep frontend-shared
lrwxr-xr-x  1 robert  staff    65B 26 Jul 08:52 frontend-shared -> /Users/robert/hypothesis/client/.yalc/@hypothesis/frontend-shared
$ ls -lh .yalc/

# Running `yarn --check-files` does replace the symlink, but doesn't touch any files in `.yalc`.
$ yarn --check-files
yarn install v1.22.10
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...

✨  Done in 14.32s.
$ ls -lh node_modules/@hypothesis/ | grep frontend-shared
drwxr-xr-x  8 robert  staff   256B 26 Jul 08:53 frontend-shared
```

Reply:
> It looks to me like yalc remove and yarn --check-files do different things. yalc remove removes the copy of the package in the project's .yalc dir, but does't appear to update the contents of node_modules/, which is what matters most. yarn --check-files ensures that the contents of node_modules/ matches yarn.lock (ie. reset all files to published packages from npm) but doesn't touch the .yalc directory.

> In other words, it doesn't look like yalc remove "restores the project-defined package version".

I wish `yalc`'s docs were a little clearer on these points (I know, I'm casting blame around a bit here).

Let's say the goal in a given local project repo (e.g. `lms`, `client`) is to make sure that project is using the package version for `@hypothesis/frontend-shared` as defined by `yarn.lock` (or `package.json`, if need be), and that all `.yalc` metadata is cleaned up. i.e. "Restore this project's dependencies, don't yalc-ify anything". 

To kind of regurgitate what I think you've written here: Sounds like `yarn --check-files` will "actually update the `node_modules` contents" to the package version in `yarn.lock`, but that leaves some `.yalc` floating around. `yalc remove` removes that `.yalc` metadata. Correct?

In which case, is it necessary, perhaps, to use _both_ commands to fully clean up?

Reply:
> In which case, is it necessary, perhaps, to use both commands to fully clean up?

Yes. However assuming that the developer only really cares about updating `node_module` to match yarn.lock, `yarn --check-files` is the only command of the two that is essential.